### ISSUE TITLE
Can stateful-cell be made to accept zero-argument functions?

### ISSUE THREAD

**Issue by jackfirth** [zyrolasting/kinda-ferpy#1]
This is the first program I wrote with `kinda-ferpy`:

```racket
#lang racket

(require kinda-ferpy)

(struct point (x y) #:transparent)

(define x (stateful-cell 1))
(define y (stateful-cell 2))
(define pos (stateful-cell (λ () (point (x) (y)))))
```

That gave me this error message:

```
../../Applications/Racket v7.5/collects/racket/private/kw.rkt:1360:47: temp5: arity mismatch;
 the expected number of arguments does not match the given number
  expected: 0
  given: 2
  arguments...:
```

Now that I've read the docs, I understand why this happened: after the discovery phase the dependencies are passed to the cell's `managed` function as arguments. I can work around that by accepting any arguments and not using them, as the examples in the docs do with `(λ _ ...)`. But if I'm not using explicit arguments, it's strange to me that I need to accept them at all. Can that be changed?

**Comment by jackfirth** [comment]
I'd be willing to take a shot at implementing this if you think it would be a good change.

**Comment by zyrolasting** [comment]
This is my miss for not calling this out in the manual at all. My apologies for that, first of all.

One of the reasons why I kept that in is to help me in code where I generate cells with dependencies that I can't write by hand in advance.

In my other project `polyglot` I have a case where I need to take an X-expression representing HTML5 and map all `href`, `src` and `srcset` attributes to requests for dependencies in a build system on its own dedicated thread. If I generate cells, it helps to know that I can permute these requests and know where they would be without writing them all out by hand. That way I can replace those same attributes with "client-ready" versions.

What I can do to maybe make everyone happy here is to allow a no-argument cell body, but 
also provide a form that helps where dependencies themselves are computed. I can toss an `->i` contract on top of that similar to the one you helped me with on Slack to enforce procedures of the same arity.

How does that sound?

**Comment by zyrolasting** [comment]
Ah, wait. I could just check your procedure arity and simply not pass anything if it's zero. Of course I can support this.

**Comment by zyrolasting** [comment]
For visitors: Disregard my long comment. Change merged in #3 does not pass arguments at all. I wasn't comfortable with the number of conditions I'd be adding with arity-based tests, and dynamic scoping gives me everything I need anyway.
